storage/store.go

```diff
@@ -292,9 +292,17 @@ func NewStore(clock *hlc.Clock, eng engine.Engine, db *client.KV, gossip *gossip
 	// Add range scanner and configure with queues.
 	s.scanner = newRangeScanner(defaultScanInterval, newStoreRangeIterator(s))
 	s.gcQueue = newGCQueue()
-	s.splitQueue = newSplitQueue(db, gossip)
 	s.verifyQueue = newVerifyQueue(s.scanner.Stats)
-	s.scanner.AddQueues(s.gcQueue, s.splitQueue, s.verifyQueue)
+	s.scanner.AddQueues(s.gcQueue, s.verifyQueue)
+	if db != nil {
```

Comment:
It breaks split_queue_test.go to put it in shouldQueue(), so I put it in process() instead. But I still have my doubts; I think it's better to disable the split queue completely than to let it run and skip the last step. Adding a check for nil in splitQueue relies on the subtlety that splitQueue.db != Store.db in these tests.